backend/pretix/db.py

```diff
@@ -30,3 +34,132 @@ def user_has_admission_ticket(email: str, event_slug: int):
         exists = cursor.fetchone()
 
     return exists[0]
+
+
+def get_user_tickets(email: str, event_slug: str):
+    if settings.SIMULATE_PRETIX_DB:
+        return []
+
+    with connections["pretix"].cursor() as cursor:
+        # fetch all the tickets that are assigned to the email
+        cursor.execute(
+            """
+            SELECT
+                tPosition.id,
+                tItem.id,
+                tItem.name,
+                tPosition.attendee_name_parts,
+                tPosition.attendee_email
+            FROM pretixbase_orderposition AS tPosition
+            LEFT JOIN pretixbase_order AS tOrder
+            ON tOrder.id = tPosition.order_id
+            LEFT JOIN pretixbase_item AS tItem
+            ON tItem.id = tPosition.item_id
+            LEFT JOIN pretixbase_event AS tEvent
+            ON tEvent.id = tOrder.event_id
+            WHERE tPosition.attendee_email = %s
+            AND tOrder.status = 'p'
+            AND tItem.admission IS TRUE
+            AND tEvent.slug = %s
+        """,
+            [email, event_slug],
+        )
+
+        tickets = cursor.fetchall()
+
+        positions_ids = []
+        items_ids = []
+
+        positions = {}
+
+        for ticket in tickets:
+            position_id = ticket[0]
+            item_id = ticket[1]
+
+            positions_ids.append(position_id)
+            items_ids.append(item_id)
+
+            positions[position_id] = {
+                "position_id": position_id,
+                "item_id": item_id,
+                "item_name": json.loads(ticket[2]),
+                "answers": [],
+            }
+
+        # This approach will only fetch the answers from the database
+        # leaving who calls the api to get (via other means) the questions data
+        # (so text, options and so on)
+        cursor.execute(
+            """
+            SELECT
+                question_id,
+                answer,
+                orderposition_id
+            FROM pretixbase_questionanswer
+            WHERE orderposition_id = ANY(%s)
+        """,
+            [positions_ids],
+        )
+
+        answers = cursor.fetchall()
+        # breakpoint()
+
+        for answer in answers:
+            position_id = answer[2]
+
+            positions[position_id]["answers"].append(
+                {"question_id": answer[0], "answer": answer[1]}
+            )
+
+        # This approach tries to fetch everything, answers AND questions list
+        # this does not support fetching a question's options/choices
+        # (those are in another table)
+        # this way allows the user to get everything without asking external services
+        # or other things
+
```

Comment:
@patrick91 I am a bit conflicted about the best way to proceed here.

What we have to do:

Fetch the user's answers for each ticket (=position) assigned to them.

How we can do it:

1. Fetch from the Pretix database all the answers of that position, without any information about the question (title, type of question, required etc.) This means that if the frontend wants to render the UI they have to fetch the questions' information. Example:

```graphql
{
  conference(code: "pycon-demo") {
    tickets(language: "en") {
      id
      questions {
        id
        required
        name
        options {
          name
          id
        }
      }
    }
  }
  
  me {
    tickets(conference: "pycon-demo", language: "en") {
      name
      positionId
      
      answers {
        questionId
        answer
      }
    }
  }
}```

This makes the backend send an HTTP request to the pretix API to get the items information and two DB requests to get the answers and user's tickets.

This is the easiest to implement (see the currently active, not commented code)

2. Another way is to fetch question+answer together (commented code), the advantage of this solution is that no API requests are being made but the SQL query is a bit of a mess (we also need to add another query to get all the possible options of a question)


I am pretty sure there are other solutions for this as well, but it might be too late for me right now 💤

Let me know what do you think @patrick91 if you can think of a better solution or anything, thanks!